firmware/Core/Inc/config/configuration.h

```diff
@@ -0,0 +1,40 @@
+#ifndef __INCLUDE_GUARD__CONFIGURATION_H
+#define __INCLUDE_GUARD__CONFIGURATION_H
+
+#include <stdint.h>
+#include <stdlib.h>
+
+typedef struct
+{
+    const char *variable_name;
+    uint32_t *num;
+    const uint8_t width_bytes; // width of the config variable (e.g., 1 for uint8_t, 2 for uint16_t, etc.)
+} CONFIG_integer_config_entry_t;
+
+typedef struct
+{
+    const char *variable_name;
+    char *variable_pointer;
+    const uint8_t max_length; // max length, including the null terminator
+} CONFIG_string_config_entry_t;
+
+extern CONFIG_integer_config_entry_t CTS1_Integer_Configuration_Variables[];
```

Comment:
Please name extern variables with standard namespaces (perhaps `CONFIG_int_config_variables` or `CONFIG_int_config_variable_table`). Same for the next few vars.